User can create poll #1

rickbacci opened the pull request:
@stevekinney

What basic goals are you looking for tomorrow. I'm going to work for a few hours and see how far I can get.

rickbacci committed:
configure mocha with expect.js

rickbacci committed:
setup basic express configuration

rickbacci committed:
create basic in-line form

rickbacci committed:
add port var to Procfile

rickbacci committed:
add port to server.js

rickbacci committed:
fix log

rickbacci committed:
fix log

rickbacci committed:
add client.js

rickbacci committed:
print poll to console

rickbacci committed:
rename button

rickbacci committed:
configure socket.io

rickbacci committed:
add random url creation

When a poll is created an object containing the
poll data is sent to the server. Socket.io picks
up the message and adds it to the polls hash.

The random urls are then sent back to the client, 
and displayed on the page.

rickbacci committed:
add admin.html to express

rickbacci committed:
fix merge conflict

stevekinney commented on public/client.js:
```diff
@@ -0,0 +1,35 @@
+  var choiceA  = document.getElementById('choice-A').value;
+  var choiceB  = document.getElementById('choice-B').value;
+  var choiceC  = document.getElementById('choice-C').value;
+  var choiceD  = document.getElementById('choice-D').value;
```
You'll probably want to take a variable number of choices.

rickbacci replied:
I can do that. I read the spec as 3-4 choices, but variable will be better.

stevekinney commented on public/client.js:
```diff
@@ -0,0 +1,35 @@
+    choiceD: choiceD,
+  }
+  console.log(poll);
+  socket.send('pollCreated', poll)
```
It might make sense to just use a regular old `POST` request.

rickbacci replied:
I thought that using ajax or sockets was the same either way. Does one have a benefit over the other?

stevekinney commented on public/client.js:
```diff
@@ -0,0 +1,35 @@
+
+var adminUrl = document.getElementById('admin-url');
+var voterUrl = document.getElementById('voter-url');
+var baseUrl  = 'http://localhost:3000'
```
You probably don't want this hard-coded in because it won't work in production.

rickbacci committed:
remove admin.html

rickbacci committed:
restructuring code

- temporarily remove socket.io from client.js
- add poll type, and end time to form
- add poll.html
- use a post request instead of sockets

rickbacci committed:
update redis

rickbacci committed:
clean up

rickbacci committed:
redisToGo configured and working on heroku

rickbacci committed:
remove un-used test page

rickbacci committed:
remove commented out code

rickbacci committed:
remove un-needed code
